apis/mapping/v1alpha1/namespaceoffloading_types.go

```diff
@@ -0,0 +1,96 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	mapsv1alpha1 "github.com/liqotech/liqo/apis/virtualKubelet/v1alpha1"
+)
+
+type mappingStrategy string
+
+const (
+	// EqualName -> remote Namespace will have the same name of local Namespace.
+	EqualName mappingStrategy = "EnforceSame"
+	// StandardName -> remote Namespace will have a default name: localNamespaceName-localClusterID.
+	StandardName mappingStrategy = "Default"
+)
+
+type deployingStrategy string
+
+const (
+	// LocalDeploy allows user only to deploy pod on local cluster.
+	LocalDeploy deployingStrategy = "Local"
+	// RemoteDeploy allows user only to deploy pod on remote clusters, possibly filtered through ClusterSelector.
+	RemoteDeploy deployingStrategy = "Remote"
+	// LocalAndRemoteDeploy allows user to deploy pod locally and remotely.
+	LocalAndRemoteDeploy deployingStrategy = "LocalAndRemote"
+)
+
+// NamespaceOffloadingSpec defines the desired state of NamespaceOffloading.
+type NamespaceOffloadingSpec struct {
+	// RemappingNameStrategy allows user to map remote Namespace name with 2 different strategy
+	// (Look at mappingStrategy).
+	// +kubebuilder:validation:Enum="EnforceSame";"Default"
+	// +kubebuilder:default="Default"
+	RemappingNameStrategy mappingStrategy `json:"remappingNameStrategy,omitempty"`
```

Comment:
```suggestion
	//  NamespaceMappingStrategy allows users to map local and remote namespace names according to two different strategies: "Default", which ensures uniqueness and prevents conflicts, and "EnforceSame", which enforces the same name at the cost of possible conflicts.
	// +kubebuilder:validation:Enum="EnforceSame";"Default"
	// +kubebuilder:default="Default"
	NamespaceMappingStrategy namespaceMappingStrategy `json:"namespaceMappingStrategy,omitempty"`
```

* The comment shoud be self-contained, otherwise it is not clear when read with kubectl explain
* I would avoid the `omitempty` keyword, to make sure it is always displayed (you need to add the `+kubebuilder:validation:Optional` marker to make the field optional

Reply:
Thank you very much